### PR TITLE
Cleanup and support web

### PR DESCRIPTION
**Pull request by dnfield**
Things I found while working on #58 

This patch drops all code related to images, and throws a slightly more meaningful exception if an image is encountered. We should figure out a better error handling strategy than that in the compiler.

It updates some stale docs that I noticed, fixes some minor formatting issues, does some light refactoring of repetitive code in the codec, and makes the example app run on web (including running the compiler in the browser).

### CODE REVIEW

File: packages/vector_graphics/lib/src/_http_io.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import 'package:flutter/foundation.dart';
+
+/// Fetches an HTTP resource from the specified [uri] using the specified [headers].
+Future<Uint8List> httpGet(Uri uri, {Map<String, String>? headers}) async {
+  final HttpClient httpClient = HttpClient();
+  final HttpClientRequest request = await httpClient.getUrl(uri);
+  if (headers != null) {
+    headers.forEach((String key, String value) {
+      request.headers.add(key, value);
+    });
```
Comment (jonahwilliams) [inline packages/vector_graphics/lib/src/_http_io.dart:13]:
```suggestion
    headers?.forEach(request.headers.add);
```

File: packages/vector_graphics/lib/src/_http_io.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  }
+  final HttpClientResponse response = await request.close();
+
+  if (response.statusCode != HttpStatus.ok) {
+    throw HttpException('Could not get network asset', uri: uri);
```
Comment (jonahwilliams) [inline packages/vector_graphics/lib/src/_http_io.dart:18]: You need to drain the stream here - see the equivalent code in flutter/flutter

File: packages/vector_graphics/lib/src/_http_web.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import 'dart:html';
+import 'dart:typed_data';
+
+/// Fetches an HTTP resource from the specified [url] using the specified [headers].
+Future<Uint8List> httpGet(Uri uri, {Map<String, String>? headers}) async {
+  final HttpRequest request = await HttpRequest.request(
+    uri.toString(),
+    requestHeaders: headers,
+  );
+  return Uint8List.fromList(utf8.encode(request.responseText!));
```
Comment (jonahwilliams) [inline packages/vector_graphics/lib/src/_http_web.dart:13]: Instead set the responseType to "arraybuffer", then `request.response` should be either a Uint8List or a List<int>

**Review by jonahwilliams — APPROVED** [review]
LGTM with nits

File: packages/vector_graphics/lib/src/_http_web.dart (hunk 3 above)
Comment (dnfield) [reply packages/vector_graphics/lib/src/_http_web.dart:13]: grumble grumble in my day you couldn't use that safely because of IE9 grumble grumble

Comment (jonahwilliams) [reply packages/vector_graphics/lib/src/_http_web.dart:13]: back in my day, we used Perl CGI and we _liked_ it!

Comment (jonahwilliams) [reply packages/vector_graphics/lib/src/_http_web.dart:13]: What is this XMLHttpRequest nonsense? AJAX will never take off!

File: packages/vector_graphics/lib/src/_http_io.dart (hunk 2 above)
Comment (dnfield) [reply packages/vector_graphics/lib/src/_http_io.dart:18]: does it really matter if we're going to throw though?

File: packages/vector_graphics/lib/src/_http_io.dart (hunk 1 above)
Comment (dnfield) [reply packages/vector_graphics/lib/src/_http_io.dart:13]: done

File: packages/vector_graphics/lib/src/_http_io.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/vector_graphics/lib/src/_http_io.dart:18]: Uhhh ... ask Zach :)

Comment (dnfield) [reply packages/vector_graphics/lib/src/_http_io.dart:18]: I see now

**Pull request merged**